### PR TITLE
Add Galileo Wrapper

### PR DESCRIPTION
**Pull request by nilsleh**
This PR takes a stab add adding a wrapper for the Galileo Model. The most difficult (or what I am not sure about is the input construction) for the encoder.

Option 1 (currently implemented):
- Create an input batch to the galileo encoder by applying their suggested [utility function](https://github.com/nasaharvest/galileo/blob/e0581f735763fa5752b1d07b378384a3f28ca697/src/data/utils.py#L36)

Option 2:
- In the training script they use a collate function to create there input batch found [here](https://github.com/nasaharvest/galileo/blob/e0581f735763fa5752b1d07b378384a3f28ca697/src/collate_fns.py#L27)

I believe that the difference in the latter is that a different random masking strategy is applied every time. I am not sure what the performance effects are for finetuning via linear-probing or partial parameters.

### CODE REVIEW

**Review by ando-shah — APPROVED** [review]

**Comment by ando-shah** [comment]
For inference, strategy #1 is probably the appropriate one
